### PR TITLE
expose more csv parser options

### LINKED ISSUE #603: expose other csv parser options in various builtin adapters
Since the `fast-csv` parser supports a few options we should expose them allowing the end user to deal with less than optimal CSV formats. These include `-comments '#', -ignoreEmpty true and -allowIncompleteRows true`

### PR DESCRIPTION
**Pull request by rlgomes**
fixes #603

exposes `comment`, `ignoreEmptyLines` and `allowIncmpleteLines` options for
csv parser along with necessary tests

### CODE REVIEW

**Comment by rlgomes** [comment]
@dmajda @dmehra @davidvgalbraith more csv options! let me know what you think of the names and what else comes to mind but I think this also handles the #557 quite well. I'll update documentation once we agree on option naming.

**Comment by rlgomes** [comment]
Other options exposed by `fast-csv` parser: https://www.npmjs.com/package/fast-csv#parsing some can be a bit "confusing" so I left them out but I'm open to suggestions.

**Comment by dmehra** [comment]
+1 
I just discovered the need to skip top lines of a csv file because they are comments (double-quoted strings), I see people do this with Python Pandas using `skiprows=N` in read_csv; however the ability to skip comments would solve the same issue for me.

**Comment by dmajda** [comment]
Typo in the commit message of https://github.com/juttle/juttle/pull/605/commits/8cc8a29d88178038b27669b0925f87aaadeb7213: “allowIncmpleteLines”.

File: lib/adapters/parsers/csv.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ class CSVParser extends base {
     constructor(options) {
         super(options);
         this.separator = options.separator ? options.separator : ',';
+        this.comment = options.comment ? options.comment : null;
```
Comment (dmajda) [inline lib/adapters/parsers/csv.js]: I wonder whether `commentChar` wouldn’t be a better name (it makes it more clear what the value is).

**Comment by dmajda** [comment]
Re https://github.com/juttle/juttle/pull/605/commits/c7bfb482d2cb894e3ac1988c567a02f0f5e6c872 — why `file` and `stdio` but not `http` and `http_server`?

File: lib/adapters/parsers/csv.js (hunk 1 above)
Comment (rlgomes) [inline lib/adapters/parsers/csv.js]: Then we'd have to rename `separator` to `separatorChar` ? I was keeping consistent with that specific option. @dmajda

**Comment by rlgomes** [comment]
@dmajda and I'll add the options to http & http_server and we can decide on the name of the option once that is done.

**Comment by rlgomes** [comment]
@dmajda did all the updates including exposing the options within the file, stdio, http and http_sever adapter and renamed `-separator` to `-separatorChar` and so we should advertise on gitter this backward comptability breakage at least to the user that requested the `separator` option.

**Comment by davidvgalbraith** [comment]
+1

**Comment by dmajda** [comment]
I’d actually keep `-separator` as it is even at the cost of slight inconsistency. Unlike `-commentChar` it will be used relatively often (so it should be short) and the term is well established.

Thinking about this, maybe rename the `-commentChar` option to `-commentStart`? That would move us from describing the type or allowed value to describing the meaning, making it more consistent with `-separator`. The cost is a slight loss of precision as people could think this could be e.g. a longer string or a regexp. Up to you.

**Comment by dmehra** [comment]
-commentSymbol ?

**Comment by rlgomes** [comment]
Alright updated with all remaining suggestions and the option is now `characterSymbol` with the `separator` option kept as is. Will merge on green.

**Pull request merged**